dht_net.go

```diff
@@ -96,6 +96,8 @@ func (dht *IpfsDHT) handleNewMessage(s network.Stream) bool {
 				ctx,
 				[]tag.Mutator{tag.Upsert(metrics.KeyMessageType, "UNKNOWN")},
 				metrics.ReceivedMessageErrors.M(1),
+				metrics.ReceivedMessages.M(1),
```

Comment:
Hm, we probably shouldn't record _either_ a received message _or_ a message error unless we receive a non-empty message. If the stream's broken for some reason, that's not something the DHT is really concerned about.

Reply:
Would it be fair to not record anything at all ? My understanding is that if the stream is broken we don't get `io.EOF` but we don't have a message either.

Reply:
The more I read your message, the more I get confused ;-). It seems to me that there is three thing we can do:
- valid message: record `metrics.ReceivedMessages.M(1)`
- broken message: record both `metrics.ReceivedMessages.M(1)` and `metrics.ReceivedMessageErrors.M(1)`
- no message: don't record anything

So what about:
- `io.EOF` --> no message
- `len(msgbytes)==0` --> no message
- `len(msgbytes)!=0` --> broken message

Reply:
I did that, let me know if that doesn't make sense.

Reply:
We might also return early on `err.Error() != "stream reset"` but that's quite brittle.

Reply:
:+1: